src/Layout/QueryParamContext.tsx

```diff
@@ -282,16 +289,65 @@ export default function QueryParamProvider({ children }: any) {
     }
   }
 
+  const loadProfile = React.useCallback(async () => {
+    const token = await getAccessTokenSilently();
+    try {
+      const profile = await apiRequest({
+        url: "/app/profile",
+        token: token,
+        setshowErrorCard,
+        tenant,
+      });
+      setprofile(profile.data);
+    } catch (err) {
+      console.error(err);
+    }
+  }, []);
+
   React.useEffect(() => {
-    if (router.query.to && country !== undefined && country !== "") {
+    if (!isLoading && isAuthenticated) {
+      loadProfile();
+    }
+  }, [isLoading, isAuthenticated, loadProfile]);
+
+  React.useEffect(() => {
+    const regex = /^pcash_/;
+    if (regex.test(router.query.to)) {
+      router.push("/");
+    } else if (router.query.to?.toString().toLowerCase() === "planetcash") {
+      if (!isLoading && !isAuthenticated) {
+        loginWithRedirect({
+          redirectUri: window?.location.href,
+        });
+      } else {
+        if (profile && profile?.planetCash?.account) {
+          loadPaymentSetup({
+            projectGUID: profile?.planetCash?.account,
+            paymentSetupCountry: country,
+            shouldSetPaymentDetails: true,
+          });
+          setdonationStep(1);
+        } else if (!profile?.planetCash) {
+          if (profile?.displayName) {
+            setprojectDetails({
+              name: `PlanetCash - ${profile?.displayName}`,
+              ownerName: profile?.displayName,
+              ownerAvatar: profile?.image,
+              purpose: "planet-cash-signup",
+            });
+          }
+          setdonationStep(4);
```

Comment:
@Shreyaschorge While functionality works fine, I do feel that we should keep a separate donation step for the planet cash signup form. Perhaps this can be kept as a future refactor.